### PR TITLE
feat: German texts/translations (#187)

### LINKED ISSUE #187: Translate to German
### What language do you want to translate the website into?

German

### What is the language code for the language?

de

### What is the locale code for the language?

de-DE

### What help will you need?

For now, no help needed.

### Implementation Checklist

- [ ] Create new site file
- [ ] Set up Netlify project
- [ ] Register new domain name
- [ ] Set up Google Analytics
- [ ] Set up Carbon Ads

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

It will probably take me several weeks to finish the translations, so if you guys are in a hurry to get this done, I may not be the right person for this task.

### PR DESCRIPTION
**Pull request by stefankolb**
This PR adds the German texts/translations for eslint.org

Couple of notes/remarks:

* I could not come up with a good/propper word for `reviewer`, so I left it in English for now. Everything that came to my mind, e.g. `Gutachter` or `Prüfer` didn't feel right. Maybe someone else has a good idea?! If not, I think we can stick with `reviewer` for now.
* I'm not very good at punctuation when it comes to commas, so there might be some errors there 🙈 

Resolves #187

### CODE REVIEW

**Comment by nzakas** [comment]
Thanks! Please be sure to sign the CLA (see [comment](https://github.com/eslint/eslint.org/pull/282#issuecomment-1197327480)). I’ll see if we can get a few more eyes on this.

File: src/_data/sites/de.yml [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+#------------------------------------------------------------------------------
+# Global Settings
+#------------------------------------------------------------------------------
+
+language:
+  code: de
+  flag: 🇩🇪
+  name: German (DE)
+locale: de-DE
+hostname: eslint.org
```
Comment (nzakas) [inline src/_data/sites/de.yml:17]:
```suggestion
hostname: de.eslint.org
```

File: src/_data/sites/de.yml [hunk 2]
```diff
@@ -0,0 +1,622 @@
...
+  name: German (DE)
+locale: de-DE
+hostname: de.eslint.org
+
+#------------------------------------------------------------------------------
+# Analytics
+#------------------------------------------------------------------------------
+
+google_analytics:
+  code: ""
```
Comment (nzakas) [inline src/_data/sites/de.yml:24]:
```suggestion
  code: "G-42E45Y2LFS"
```

File: src/_data/sites/de.yml [hunk 3]
```diff
@@ -0,0 +1,622 @@
...
+google_analytics:
+  code: "G-42E45Y2LFS"
+
+#------------------------------------------------------------------------------
+# Ads
+#------------------------------------------------------------------------------
+
+carbon_ads:
+  serve:
+  placement:
```
Comment (nzakas) [inline src/_data/sites/de.yml:32]:
```suggestion
  serve: ""
  placement: ""
```

**Review by nzakas — APPROVED** [review]
Sorry I lost track of this! Building the site now.

**Comment by stefankolb** [comment]
> Sorry I lost track of this! Building the site now.

No worries, glad I could be of help and cool to see it went live

**Comment by nzakas** [comment]
@stefankolb We'd like to pay you for this contribution but I couldn't find an email address to contact you. Can you please shoot us an email at contact (at) eslint (dot) org? Thanks!

**Comment by stefankolb** [comment]
That's awesome, I just sent an email. Thank you!

**Pull request merged**